engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerOverseer.java

```diff
@@ -0,0 +1,107 @@
+package com.odysseusinc.arachne.executionengine.execution.r;
+
+import static com.odysseusinc.arachne.execution_engine_common.api.v1.dto.AnalysisRequestTypeDTO.R;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.async.ResultCallback;
+import com.github.dockerjava.api.command.LogContainerCmd;
+import com.github.dockerjava.api.command.WaitContainerResultCallback;
+import com.github.dockerjava.api.exception.NotFoundException;
+import com.github.dockerjava.api.model.Frame;
+import com.odysseusinc.arachne.execution_engine_common.api.v1.dto.AnalysisRequestTypeDTO;
+import com.odysseusinc.arachne.execution_engine_common.api.v1.dto.ExecutionOutcome;
+import com.odysseusinc.arachne.execution_engine_common.api.v1.dto.Stage;
+import com.odysseusinc.arachne.executionengine.execution.AbstractOverseer;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+@Slf4j
+@Getter
+public class DockerOverseer extends AbstractOverseer {
+    private final ScheduledExecutorService executor = new ScheduledThreadPoolExecutor(1) {{
+        setRemoveOnCancelPolicy(true);
+    }};
+    private final CompletableFuture<String> init;
+
+    private volatile int pos;
+    private final DockerClient client;
+
+    public DockerOverseer(
+            long id, DockerClient client, Instant started, int timeout, StringBuffer stdout, CompletableFuture<String> init,
+            int updateInterval, BiConsumer<String, String> callback, String image, int killTimeout
+    ) {
+        super(id, callback, started, image, killTimeout, new StringBuffer(), init.handle((containerId, throwable) -> {
+            if (throwable != null) {
+                String out = stdout.append("\n").append(ExceptionUtils.getStackTrace(throwable)).toString();
```

Comment:
Not sure what you mean. That "\n" is carriage return, right?

Reply:
"
    \r = CR (Carriage Return) → Used as a new line character in Mac OS before X
    \n = LF (Line Feed) → Used as a new line character in Unix/Mac OS X
    \r\n = CR + LF → Used as a new line character in Windows
"

Reply:
Interesting, I have somehow lived my whole java life with the misconception that "\n" and "\r" meanings are opposite.
AFAIK, almost any text viewer/editor with be happy enough with a single LF, regardless of platform.
However since SQLService is already using "\r\n" for logs, I guess it's safer just to follow that suit.
Fixed in AnalysisService as well.